ai.finto.fi/static/js/annif.js

```diff
@@ -84,45 +92,96 @@ const mainApp = createApp({
       show_alert_file_format: false,
       show_alert_request_failed: false,
       show_alert_request_failed_url: false,
+      show_alert_language_detection_failed: false,
       show_dragging_effect: false,
       supported_formats: ['txt', 'pdf', 'doc', 'docx', 'odt', 'rtf', 'pptx', 'epub', 'html']
     }
   },
+  provide() {
+    return {
+      detectLanguage: this.detectLanguage
+    };
+  },
+  watch: {
+    text_language(newValue, oldValue) {
+      if (newValue != null) {
+        this.show_alert_language_detection_failed = false;
+      }
+    },
+  },
   methods: {
     clear() {
       // clear previous text, suggestions, alerts etc
       this.text = ''
       this.show_results = false
       this.selected_file = ''
       this.selected_url = ''
+      this.is_language_detected = false;
       this.show_alert_file_size = false
       this.show_alert_file_format = false
       this.show_alert_request_failed = false
       this.show_alert_request_failed_url = false
+      this.show_alert_language_detection_failed = false
+    },
+    detectLanguage() {
+      if (this.text.length > 9) {
+        this.detecting_language = true;
+        this.text_language = 'none';
+        fetch(annif_base_url + 'detect-language', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            text: this.text,
+            languages: ["fi", "sv", "en"]  // TODO Here should be only langs that selected project supports
```

Comment:
Is the TODO a problem in practice for current Finto AI? For some vocabs, we don't support all languages.

Reply:
Now, when e.g. KAUNO vocab is selected and English text is inputted, language detection chooses English (manually it is not selectable), and Get subjects suggestions button is clickable. When clicked, no suggestions appear (console logs `TypeError: results is undefined`): 

![image](https://github.com/user-attachments/assets/f3cfc9db-5d62-460a-82fe-8f6afb0a8c0f)

This is not ideal.

Reply:
Actually, if e.g. English was selected for text language and vocab selection switched to e.g. KAUNO, there was a similar situation (an unsupported vocab - language combination) as when the language detection switched the language automatically.

To resolve this, I made "Get subject suggestions" button disabled when an unsupported vocab - language combination is selected (what ever the cause).